modules/apps/feature-flag/feature-flag-web/src/main/resources/META-INF/resources/js/FeatureFlagToggle.tsx

```diff
@@ -55,6 +55,14 @@ const FeatureFlagToggle = ({
 						? data.dependentFeatureFlags
 						: []),
 				]);
+
+				const toggleButton = document.querySelector(
+					`#${featureFlagKey}`
+				) as HTMLElement;
+
+				setTimeout(() => {
+					toggleButton?.focus();
+				}, 10);
```

Comment:
You can then add a TODO comment here and then look at this with other solutions, such as trying to use `useEffect` or `useLayoutEffect` to maintain focus...

Reply:
Was there additional context to this comment that's now gone? setTimeout does look odd here though. Seems like there should be another solution.

Reply:
Yeah, for some reason the focus is being lost which is very strange we tried to make some assumptions but without much success, so we went with a palliative to solve this problem for now.

Yes, there is another solution here which would be to use `useEffect`, we need to ensure that we call `element.focus()` after the DOM update happens as the call to `onItemsChange` will still schedule an update and this is asynchronous so we added `setTimeout` as a stopgap here for now but it would be possible to do this via useEffect also to avoid `setTimeout`.

Reply:
Hey @ethib137 
I've tried  implementing with `useEffect`, `useLayoutEffect` and `flushSync`  but had no success.

Reply:
So what's happening here is that when we make the request to update the feature flag we then disable the toggle while the request is taking place so that multiple simultaneous requests can't be initiated for the same feature flag. When we disable the toggle, focus is lost because a disabled element cannot receive focus. 

So, what we want to do is when the request is complete and the toggle is no longer disabled then put the focus back on the toggle. We can do this like this:
```
useEffect(() => {
	if (!disabled && !isLoading) {
		toggleRef.current.focus();
	}
}, [disabled, isLoading]);
```

I've added this in the commit [here](https://github.com/liferay/liferay-portal/commit/cb24b4787501f4a79d1b2355961c70b1828d2e3f). I don't have permission to push to your branch @ilzamcmed so you can add me as a contributor or just cherry pick my commit. 

Let me know if you see any problems with this approach.

Reply:
Thanks @ethib137! :tada:  
I'll make a cherry pick to go with the approach you suggested.
 :clap: :clap: :clap:

Reply:
Hey @ethib137, I found an issue with this implementation while testing. 
When we enter the page, focus go straight to the last toggle button.

Reply:
Good catch @ilzamcmed! Here is another commit to handle that. We can set a separate initial state for isLoading and then check whether that initial state is still set before moving focus: https://github.com/liferay/liferay-portal/commit/7fe7887054aef872e4e3ee05d1f05bc5806cf181